sdcm/cluster_aws.py

```diff
@@ -138,6 +138,12 @@ def _create_on_demand_instances(self, count, interfaces, ec2_user_data, dc_idx=0
         if instance_profile:
             params['IamInstanceProfile'] = {'Name': instance_profile}
         instances = self._ec2_services[dc_idx].create_instances(**params)
+
+        ec2 = ec2_client.EC2ClientWrapper(region_name=self.region_names[dc_idx])
+        instance_ids = [instance.instance_id for instance in instances]
```

Comment:
you are talking about the list comprehension ? or the creation of `EC2ClientWrapper` ?

the add_tags isn't aware of `Instance`, and in some cases I have only ids, and in some cases Instances.

Reply:
Yes, I am talking about the list comprehension. And the `add_tags` method doesn't need to know the specific class. It may just be aware of the method that must be called in one of 2 possible cases. It is easy situation.